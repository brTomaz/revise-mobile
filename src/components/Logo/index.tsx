import React from 'react'
import { Image } from 'react-native'

import logo from '@/assets/images/logo.png'
import logoFirstLetter from '@/assets/images/logo-first-letter.png'

type LogoProps = {
  minimal?: boolean
}

const Logo = ({ minimal = false }: LogoProps) =>
  minimal ? (
    <Image source={logoFirstLetter} testID="minimal-logo" />
  ) : (
    <Image source={logo} testID="logo" />
  )

export default Logo
