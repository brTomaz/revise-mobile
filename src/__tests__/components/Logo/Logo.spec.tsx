import React from 'react'
import { render } from '@testing-library/react-native'

import Logo from '@/components/Logo'

describe('<Logo />', () => {
  it('should render a complete Logo by default', () => {
    const { getByTestId } = render(<Logo />)

    const logo = getByTestId('logo')

    expect(logo).toBeTruthy()
    expect(logo).toMatchSnapshot()
  })

  it('should render a minimal Logo when this property is passed', () => {
    const { getByTestId } = render(<Logo minimal />)

    const minimalLogo = getByTestId('minimal-logo')

    expect(minimalLogo).toBeTruthy()
  })
})
