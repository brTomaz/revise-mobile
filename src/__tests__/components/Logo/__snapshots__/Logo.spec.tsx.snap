// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> should render a complete Logo by default 1`] = `
<Image
  source={
    Object {
      "testUri": "../../../src/assets/images/logo.png",
    }
  }
  testID="logo"
/>
`;
